pkg/verifier/cosign/trustpolicy.go

```diff
@@ -0,0 +1,273 @@
+/*
+Copyright The Ratify Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cosign
+
+import (
+	"context"
+	"crypto"
+	"fmt"
+	"os"
+
+	re "github.com/deislabs/ratify/errors"
+	"github.com/deislabs/ratify/internal/constants"
+	"github.com/deislabs/ratify/pkg/keymanagementprovider"
+	vu "github.com/deislabs/ratify/pkg/verifier/utils"
+	"github.com/deislabs/ratify/utils"
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/fulcio"
+	"github.com/sigstore/cosign/v2/cmd/cosign/cli/rekor"
+	"github.com/sigstore/cosign/v2/pkg/cosign"
+	"github.com/sigstore/sigstore/pkg/cryptoutils"
+)
+
+type KeyConfig struct {
+	Provider string `json:"provider,omitempty"`
+	Name     string `json:"name,omitempty"`
+	Version  string `json:"version,omitempty"`
+	File     string `json:"file,omitempty"`
+}
+
+type KeylessConfig struct {
+	RekorURL    string `json:"rekorURL,omitempty"`
+	CTLogVerify *bool  `json:"ctLogVerify,omitempty"`
+}
+
+type TrustPolicyConfig struct {
+	Name       string        `json:"name"`
```

Comment:
should TrustPolicy have a version incase of schema change?

Reply:
sure I can add that. good idea

Reply:
I added a new `version` property to the `TrustPolicyConfig` this is the config that relates to each item in the `trustPolicies` array. Here's a sample: 

```yaml
apiVersion: config.ratify.deislabs.io/v1beta1
kind: Verifier
metadata:
  name: verifier-cosign
  annotations:
    helm.sh/hook: pre-install,pre-upgrade
    helm.sh/hook-weight: "5"
spec:
  name: cosign
  artifactTypes: application/vnd.dev.cosign.artifact.sig.v1+json
  parameters:
    trustPolicies:
      - name: default
        version: 1.0.0 <----- NEW ADDITION
        scopes:
          - "*"
        keys:
          - provider: gatekeeper-system/kmprovider-akv
```

For now, there's only one version `1.0.0` supported. But in future if we introduce new version, I'm thinking then we can add the multi version selection logic then? (unless you feel we should do that now). There are checks to make sure that only the supported version is provided. And for now if the user omits the version, it assumes the default version of '1.0.0' cc: @susanshi

Reply:
thankyou for adding this Akash. We should document what is the default verison, once we have the next version,  will the default be the newest supported version? or the oldest supported version.  Defaulting to 1.0.0 is good for this release.  Can you clarify what is the multi version selection logic? is it selection trust Policies with version we support?

Reply:
Yes the intention would be that the default version would be the latest version so it would change if we updated the config version. for "multi version selection logic" this is the functionality so that we invoke different trust policy behavior based on the version provided. Currently, the version is simply verified to make sure it is within a supported array of versions. There's no "branching" of implementation based on the version